Assets/Scripts/Layers/PopulateAddLayerPanel.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Netherlands3D.Twin.Layers;
+using Netherlands3D.Twin.Projects;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Netherlands3D.Twin
+{
+    [RequireComponent(typeof(AddLayerPanel))]
+    public class PopulateAddLayerPanel : MonoBehaviour
+    {
+        private AddLayerPanel addLayerPanel;
+
+        [Header("Panel references")]
+        [SerializeField] private Button addLayerMenuButton;
+        [SerializeField] private Transform titleRow;
+        [SerializeField] private Transform mainButtonPanel;
+        [SerializeField] private Transform contentParent;
+        
+        [Header("Prefab references")]
+        [SerializeField] private GameObject titlePrefab;
+        [SerializeField] private Button panelButtonPrefab;
+        [SerializeField] private GameObject groupPanelPrefab;
+        [SerializeField] private ObjectLibraryButton buttonPrefab;
+        private void Awake()
+        {
+            addLayerPanel = GetComponent<AddLayerPanel>();
+        }
+
+        private void Start()
+        {
+            Populate(ProjectData.Current.PrefabLibrary);
+        }
+
+        private void Populate(PrefabLibrary library)
+        {
+            foreach (var group in library.prefabGroups)
+            {
+                if (!group.autoPopulateUI)
+                    continue;
+                
+                var groupPanel = CreateGroupPanel(group.groupName);
+                foreach (var prefab in group.prefabs)
+                {
+                    CreateButton(prefab, groupPanel);
+                }
+            }
+        }
+
+        private GameObject CreateGroupPanel(string groupGroupName)
```

Comment:
Had de inhoud hiervan niet beter in een GroupPanel controller script gepast, zodat de GroupPanel voor zichzelf verantwoordelijk is, en zodat je in de prefab referentie een GroupPanel class reference kan opnemen ipv GameObject en daarmee het net iets typestricter wordt?

Reply:
Dat is geen slecht idee, maar ik denk dat het beter een aparte story/pr kan worden. Dit was bedoeld als eerste opzet om ervoor te zorgen dat alleen prefabs/objecten in het AddLayerPanel terecht komen die ook weer ingeladen kunnen worden. Het is nog niet 100% gekoppeld, omdat bijvoorbeeld de Standaardlagen allemaal een functionality listener hebben en ik daar nu niet een oplossing voor wilde bedenken omdat het anders de scope van deze story te groot zou maken. die koppeling compleet maken lijkt me wel een goed idee voor een IP sprint